### PR TITLE
doc: Generate CHANGELOG

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s):

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,73 @@
 # Changelog
 
+## [v1.12.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.12.2) (October 17, 2023)
```
Comment (lantoli) [inline CHANGELOG.md]: I believe this is not the date format @Zuhairahmed likes, we might want to change date-format=%B %d, %Y.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,73 @@
...
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.12.1...v1.12.2)
+
+**Enhancements**
+
+- Supports new data expiration rule attribute in online archive [\#1528](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1528) ([AgustinBettati](https://github.com/AgustinBettati))
+- Supports new notifier\_id attribute in alert configuration notifications [\#1514](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1514) ([AgustinBettati](https://github.com/AgustinBettati))
+
+**Bug Fixes**
+
+- Encryption at rest return inconsistent plan when setting secret access key [\#1529](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1529) ([andreaangiolillo](https://github.com/andreaangiolillo))
```
Comment (lantoli) [inline CHANGELOG.md]: returnS (i suppose it's because the PR title)

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,73 @@
...
+- Migrates online archive resource and data sources to new SDK [\#1523](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1523) ([AgustinBettati](https://github.com/AgustinBettati))
+- Bump github.com/hashicorp/terraform-plugin-framework from 1.4.0 to 1.4.1 [\#1521](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1521) ([dependabot[bot]](https://github.com/apps/dependabot))
+- Bump github.com/zclconf/go-cty from 1.14.0 to 1.14.1 [\#1520](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1520) ([dependabot[bot]](https://github.com/apps/dependabot))
+- Bump github.com/gruntwork-io/terratest from 0.45.0 to 0.46.0 [\#1519](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1519) ([dependabot[bot]](https://github.com/apps/dependabot))
+- Bump github.com/aws/aws-sdk-go from 1.45.21 to 1.45.24 [\#1518](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1518) ([dependabot[bot]](https://github.com/apps/dependabot))
+- Bump github.com/hashicorp/hcl/v2 from 2.18.0 to 2.18.1 [\#1517](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1517) ([dependabot[bot]](https://github.com/apps/dependabot))
```
Comment (lantoli) [inline CHANGELOG.md]: knitpit: I'd probably group all bumps and move them to the end of the section

**Review by lantoli — APPROVED** [review]

File: CHANGELOG.md (hunk 3 above)
Comment (andreaangiolillo) [reply CHANGELOG.md]: Updated

File: CHANGELOG.md (hunk 1 above)
Comment (andreaangiolillo) [reply CHANGELOG.md]: updated

File: CHANGELOG.md (hunk 2 above)
Comment (andreaangiolillo) [reply CHANGELOG.md]: updated

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,73 @@
...
+- Fixes cleanup-test-env script continues if delete of one project fails [\#1516](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1516) ([AgustinBettati](https://github.com/AgustinBettati))
+- Updates atlas-sdk to v20231001001 [\#1515](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1515) ([AgustinBettati](https://github.com/AgustinBettati))
+- Adjusts time for stale github issues to close after 1 week of inactivity [\#1512](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1512) ([AgustinBettati](https://github.com/AgustinBettati))
+- Updates 1.10.0-upgrade-guide.html.markdown [\#1511](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1511) ([Zuhairahmed](https://github.com/Zuhairahmed))
+- Updates template issue with clearer guidelines. [\#1510](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1510) ([marcosuma](https://github.com/marcosuma))
+- Avoid including provider\_disk\_type\_name property in cluster update request if attribute was removed [\#1508](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1508) ([AgustinBettati](https://github.com/AgustinBettati))
```
Comment (AgustinBettati) [inline CHANGELOG.md]: We can move this into the Bug Fix sections (PR #1508)

File: .github_changelog_generator [hunk 5]
```diff
@@ -1,4 +1,4 @@
-future-release=v1.12.1
-since-tag=v1.12.0
+future-release=v1.12.2
+since-tag=v1.12.1
 date-format=%B %d, %Y
```
Comment (lantoli) [inline .github_changelog_generator]:
```suggestion
date-format=%Y-%m-%d
```

Comment (andreaangiolillo) [reply .github_changelog_generator]: updated

File: CHANGELOG.md (hunk 4 above)
Comment (andreaangiolillo) [reply CHANGELOG.md]: updated

File: .github_changelog_generator [hunk 6]
```diff
@@ -1,4 +1,4 @@
-future-release=v1.12.1
-since-tag=v1.12.0
-date-format=%B %d, %Y
+future-release=v1.12.2
+since-tag=v1.12.1
+date-format=%Y %m, %d
```
Comment (lantoli) [inline .github_changelog_generator]: better use date-format=%Y-%m-%d to have like 2023-10-17

Using %Y %m, %d will be like 2023 10, 17

I found that the tool is using: https://apidock.com/ruby/DateTime/strftime

Comment (andreaangiolillo) [reply .github_changelog_generator]: it makes sense. Updating

Comment (andreaangiolillo) [reply .github_changelog_generator]: updated

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,5 +1,73 @@
...
+
+- Encryption at rest returns inconsistent plan when setting secret access key [\#1529](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1529) ([andreaangiolillo](https://github.com/andreaangiolillo))
+- Alert configuration data source nil pointer with third party notifications [\#1513](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1513) ([AgustinBettati](https://github.com/AgustinBettati))
+- Adjusts format of database user resource id as defined in previous versions [\#1506](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1506) ([AgustinBettati](https://github.com/AgustinBettati))
+- Removes delete partition\_fields statements. [\#1499](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1499) ([marcosuma](https://github.com/marcosuma))
+- Changes validation of empty provider credentials from Error to Warning [\#1501](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1501) ([AgustinBettati](https://github.com/AgustinBettati))
+- Use container\_id from created cluster in example [\#1475](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1475) ([mckornfield](https://github.com/mckornfield))
```
Comment (davidhou17) [inline CHANGELOG.md]:
```suggestion
- Uses `container_id` from created cluster in example [\#1475](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1475) ([mckornfield](https://github.com/mckornfield))
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -1,5 +1,73 @@
...
+- Adjusts time for stale github issues to close after 1 week of inactivity [\#1512](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1512) ([AgustinBettati](https://github.com/AgustinBettati))
+- Updates 1.10.0-upgrade-guide.html.markdown [\#1511](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1511) ([Zuhairahmed](https://github.com/Zuhairahmed))
+- Updates template issue with clearer guidelines. [\#1510](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1510) ([marcosuma](https://github.com/marcosuma))
+- Avoid including provider\_disk\_type\_name property in cluster update request if attribute was removed [\#1508](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1508) ([AgustinBettati](https://github.com/AgustinBettati))
```
Comment (davidhou17) [inline CHANGELOG.md]:
```suggestion
- Avoids including `provider_disk_type_name` property in cluster update request if attribute was removed [\#1508](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1508) ([AgustinBettati](https://github.com/AgustinBettati))
```

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,5 +1,73 @@
...
+- Supports new notifier\_id attribute in alert configuration notifications [\#1514](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1514) ([AgustinBettati](https://github.com/AgustinBettati))
+
+**Bug Fixes**
+
+- Encryption at rest returns inconsistent plan when setting secret access key [\#1529](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1529) ([andreaangiolillo](https://github.com/andreaangiolillo))
+- Alert configuration data source nil pointer with third party notifications [\#1513](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1513) ([AgustinBettati](https://github.com/AgustinBettati))
+- Adjusts format of database user resource id as defined in previous versions [\#1506](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1506) ([AgustinBettati](https://github.com/AgustinBettati))
+- Removes delete partition\_fields statements. [\#1499](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1499) ([marcosuma](https://github.com/marcosuma))
```
Comment (davidhou17) [inline CHANGELOG.md]: [nit] Removes period to be consistent with rest of list
```suggestion
- Removes delete `partition_fields` statements [\#1499](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1499) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 10]
```diff
@@ -1,5 +1,73 @@
...
+- Changes validation of empty provider credentials from Error to Warning [\#1501](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1501) ([AgustinBettati](https://github.com/AgustinBettati))
+- Use container\_id from created cluster in example [\#1475](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1475) ([mckornfield](https://github.com/mckornfield))
+- Adjusts time for stale github issues to close after 1 week of inactivity [\#1512](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1512) ([AgustinBettati](https://github.com/AgustinBettati))
+- Updates 1.10.0-upgrade-guide.html.markdown [\#1511](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1511) ([Zuhairahmed](https://github.com/Zuhairahmed))
+- Updates template issue with clearer guidelines. [\#1510](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1510) ([marcosuma](https://github.com/marcosuma))
```
Comment (davidhou17) [inline CHANGELOG.md]:
```suggestion
- Updates template issue with clearer guidelines [\#1510](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1510) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,5 +1,73 @@
...
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.12.1...v1.12.2)
+
+**Enhancements**
+
+- Supports new data expiration rule attribute in online archive [\#1528](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1528) ([AgustinBettati](https://github.com/AgustinBettati))
+- Supports new notifier\_id attribute in alert configuration notifications [\#1514](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1514) ([AgustinBettati](https://github.com/AgustinBettati))
+
+**Bug Fixes**
+
+- Encryption at rest returns inconsistent plan when setting secret access key [\#1529](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1529) ([andreaangiolillo](https://github.com/andreaangiolillo))
```
Comment (davidhou17) [inline CHANGELOG.md]: Is this a description of the bug? If so, let's update to clarify the actual fix. Maybe something like:

```suggestion
- Fixes issue where Encryption at rest returns inconsistent plan when setting secret access key [\#1529](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1529) ([andreaangiolillo](https://github.com/andreaangiolillo))
```

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,5 +1,73 @@
...
+- Alert configuration data source nil pointer with third party notifications [\#1513](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1513) ([AgustinBettati](https://github.com/AgustinBettati))
```
Comment (davidhou17) [inline CHANGELOG.md]: Same comment as above
```suggestion
- Fixes issue where alert configuration data source for third party notifications returns nil pointer [\#1513](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1513) ([AgustinBettati](https://github.com/AgustinBettati))
```

File: CHANGELOG.md [hunk 13]
```diff
@@ -1,5 +1,73 @@
...
+- `mongodbatlas_project_ip_access_list` causes invalid plans [\#1484](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1484)
+- Changes to oplog\_min\_retention\_hours not being applied when set to null [\#1481](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1481)
+- Create alert with more than 1 notification [\#1473](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1473)
+
+**Internal Improvements**
+
+- Migrates online archive resource and data sources to new SDK [\#1523](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1523) ([AgustinBettati](https://github.com/AgustinBettati))
+- Fixes cleanup-test-env script continues if delete of one project fails [\#1516](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1516) ([AgustinBettati](https://github.com/AgustinBettati))
+- Updates atlas-sdk to v20231001001 [\#1515](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1515) ([AgustinBettati](https://github.com/AgustinBettati))
+- Module naming convention [\#1500](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1500) ([govardhanpagidi](https://github.com/govardhanpagidi))
```
Comment (davidhou17) [inline CHANGELOG.md]: [nit] Adding a descriptive verb to align with rest of list
```suggestion
- Fixes module naming convention [\#1500](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1500) ([govardhanpagidi](https://github.com/govardhanpagidi))
```

File: CHANGELOG.md [hunk 14]
```diff
@@ -1,5 +1,73 @@
...
+- Updates cluster update handler to update advanced\_configuration first and make oplog\_min\_retention\_hours non-computed [\#1497](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1497) ([maastha](https://github.com/maastha))
+- Adds coverage report to PRs [\#1496](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1496) ([andreaangiolillo](https://github.com/andreaangiolillo))
+- Adds sagemaker quickstart to repo [\#1494](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1494) ([govardhanpagidi](https://github.com/govardhanpagidi))
+- Closes code block in "Resource: Cloud Provider Access Configuration Paths" documentation page [\#1487](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1487) ([mattt416](https://github.com/mattt416))
+- Bump github.com/gruntwork-io/terratest from 0.43.13 to 0.44.0 [\#1482](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1482) ([dependabot[bot]](https://github.com/apps/dependabot))
```
Comment (davidhou17) [inline CHANGELOG.md:58]: [nit] This applies to the remaining occurrences of "bump" as well
 
```suggestion
- Bumps github.com/gruntwork-io/terratest from 0.43.13 to 0.44.0 [\#1482](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1482) ([dependabot[bot]](https://github.com/apps/dependabot))
```

**Review by davidhou17 — COMMENTED** [review]
Back to you with a handful of small suggestions. LGTM after those fixes!

File: CHANGELOG.md [hunk 15]
```diff
@@ -1,5 +1,73 @@
 # Changelog
 
+## [v1.12.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.12.2) (2023-10-17)
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
## [v1.12.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.12.2) (2023-10-18)
```

File: CHANGELOG.md [hunk 16]
```diff
@@ -1,5 +1,73 @@
...
+- Updates 1.10.0-upgrade-guide.html.markdown [\#1511](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1511) ([Zuhairahmed](https://github.com/Zuhairahmed))
+- Updates template issue with clearer guidelines. [\#1510](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1510) ([marcosuma](https://github.com/marcosuma))
+- Avoid including provider\_disk\_type\_name property in cluster update request if attribute was removed [\#1508](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1508) ([AgustinBettati](https://github.com/AgustinBettati))
+
+**Deprecations and Removals**
+
+- Removes the data source mongodbatlas_privatelink_endpoint_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
+- Removes the data source mongodbatlas_privatelink_endpoints_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
+- Removes mongodbatlas_privatelink_endpoint_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md:29]:
```suggestion
```

File: CHANGELOG.md [hunk 17]
```diff
@@ -1,5 +1,73 @@
...
+- Adjusts time for stale github issues to close after 1 week of inactivity [\#1512](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1512) ([AgustinBettati](https://github.com/AgustinBettati))
+- Updates 1.10.0-upgrade-guide.html.markdown [\#1511](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1511) ([Zuhairahmed](https://github.com/Zuhairahmed))
+- Updates template issue with clearer guidelines. [\#1510](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1510) ([marcosuma](https://github.com/marcosuma))
+- Avoid including provider\_disk\_type\_name property in cluster update request if attribute was removed [\#1508](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1508) ([AgustinBettati](https://github.com/AgustinBettati))
+
+**Deprecations and Removals**
+
+- Removes the data source mongodbatlas_privatelink_endpoint_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Removes the data source `mongodbatlas_privatelink_endpoint_service_adl` [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 18]
```diff
@@ -1,5 +1,73 @@
...
+- Removes the data source mongodbatlas_privatelink_endpoints_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Removes the data source `mongodbatlas_privatelink_endpoints_service_adl` [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 19]
```diff
@@ -1,5 +1,73 @@
...
+- Removes mongodbatlas_privatelink_endpoint_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
+- Removes the resource mongodbatlas_privatelink_endpoints_service_adl [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Removes the resource `mongodbatlas_privatelink_endpoints_service_adl` [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 20]
```diff
@@ -1,5 +1,73 @@
...
+- Removes the data source mongodbatlas_data_lake [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Removes the data source `mongodbatlas_data_lake` [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 21]
```diff
@@ -1,5 +1,73 @@
...
+- Removes the data source mongodbatlas_data_lakes [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Removes the data source `mongodbatlas_data_lakes` [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 22]
```diff
@@ -1,5 +1,73 @@
...
+- Removes the resource mongodbatlas_data_lake [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Removes the resource `mongodbatlas_data_lake` [\#1503](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1503) ([marcosuma](https://github.com/marcosuma))
```

File: CHANGELOG.md [hunk 23]
```diff
@@ -1,5 +1,73 @@
...
+
+**Closed Issues**
+
+- Error changing user [\#1509](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1509)
+- Provider "registry.terraform.io/mongodb/mongodbatlas" planned an invalid value [\#1498](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1498)
+- IP allowlist range force replacement on 1.12.0 [\#1495](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1495)
+- Importing Online Archive resources is missing parameter partition\_fields in terraform state [\#1492](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1492)
+- mongodbatlas\_network\_container faulty optional variable regions [\#1490](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1490)
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- `mongodbatlas_network_container` faulty optional variable regions [\#1490](https://github.com/mongodb/terraform-provider-mongodbatlas/issues/1490)
```

File: CHANGELOG.md [hunk 24]
```diff
@@ -1,5 +1,73 @@
 # Changelog
 
+## [v1.12.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.12.2) (2023-10-17)
+
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.12.1...v1.12.2)
+
+**Enhancements**
+
+- Supports new data expiration rule attribute in online archive [\#1528](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1528) ([AgustinBettati](https://github.com/AgustinBettati))
+- Supports new notifier\_id attribute in alert configuration notifications [\#1514](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1514) ([AgustinBettati](https://github.com/AgustinBettati))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Supports new `notifier_id` parameter in `mongodbatlas_alert_configuration` [\#1514](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1514) ([AgustinBettati](https://github.com/AgustinBettati))
```

File: CHANGELOG.md [hunk 25]
```diff
@@ -1,5 +1,73 @@
 # Changelog
 
+## [v1.12.2](https://github.com/mongodb/terraform-provider-mongodbatlas/tree/v1.12.2) (2023-10-17)
+
+[Full Changelog](https://github.com/mongodb/terraform-provider-mongodbatlas/compare/v1.12.1...v1.12.2)
+
+**Enhancements**
+
+- Supports new data expiration rule attribute in online archive [\#1528](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1528) ([AgustinBettati](https://github.com/AgustinBettati))
```
Comment (Zuhairahmed) [inline CHANGELOG.md]:
```suggestion
- Supports `data_expiration_rule` parameter in 'mongodbatlas_online_archive` [\#1528](https://github.com/mongodb/terraform-provider-mongodbatlas/pull/1528) ([AgustinBettati](https://github.com/AgustinBettati))
```

**Review by Zuhairahmed — APPROVED** [review]
LGTM, few minor nits. Let me know if you need to chat on anything. Great work @AgustinBettati / @andreaangiolillo !

**Review by marcosuma — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

**Pull request merged**